### DISCUSSION TITLE
Why does searching using "$eq" and putting an array as value give a type error?

### DISCUSSION THREAD

**Discussion by sawa-ko** [mikro-orm/mikro-orm#2958] [Q&A] [answered]
I am trying to do the following query but typescript says there is a type error.
I try the same query in the mongodb console and it works perfectly, I don't understand why it says there is an error.

> Code

![image](https://user-images.githubusercontent.com/56084970/160262748-3928350e-d7d8-45d9-89f3-9e1412988e3b.png)

> Error

![image](https://user-images.githubusercontent.com/56084970/160262753-4fca6f52-44a0-408b-8be8-9a1a27d389d1.png)

> Entity field
```
@ManyToMany(() => UserEntity)
participants = new Collection<UserEntity>(this);
```

> Mongo shell

![image](https://user-images.githubusercontent.com/56084970/160262783-f8c4549a-759b-4632-953c-edd623adbc4d.png)

**Comment by B4nan** [comment, marked as answer]
`$eq` is not an array operator in the ORM context, it expects a single scalar value on the right hand side. First check if it's just a typings issue by casting the query `as any` to get around this.

**Reply by sawa-ko** [reply]
Yes, it's a typing error, because if I put `as any` the problem is solved, I guess the mongoDb EntityRepository should accept an array in the `$eq` operator.
